packages/proxy/lib/http/response-middleware.ts

```diff
@@ -381,11 +382,82 @@ const MaybePreventCaching: ResponseMiddleware = function () {
   this.next()
 }
 
+const determineIfNeedsMultiDomainHandling = (ctx) => {
+  const previousAUTRequestUrl = ctx.getPreviousAUTRequestUrl()
+
+  // A cookie needs multi-domain handling if it's an AUT request and
+  // either the request itself is cross-origin or the origins between
+  // requests don't match, since the browser won't set them in that
+  // case and if it's secondary-domain -> primary-domain, we don't
+  // recognize the request as cross-origin
+  return (
+    !!ctx.req.isAUTFrame &&
+    (
+      (previousAUTRequestUrl && !cors.urlOriginsMatch(previousAUTRequestUrl, ctx.req.proxiedUrl))
+      || !reqMatchesOriginPolicy(ctx.req, ctx.getRemoteState())
+    )
+  )
+}
+
+const cookieSameSiteRegex = /SameSite=(\w+)/i
+const cookieSecureRegex = /Secure/i
+const cookieSecureSemicolonRegex = /;\s*Secure/i
+
+const ensureSameSiteNone = ({ cookie, browser, isLocalhost, url }) => {
+  debug('original cookie: %s', cookie)
+
+  if (cookieSameSiteRegex.test(cookie)) {
+    debug('change cookie to SameSite=None')
+    cookie = cookie.replace(cookieSameSiteRegex, 'SameSite=None')
+  } else {
+    debug('add SameSite=None to cookie')
+    cookie += '; SameSite=None'
+  }
+
+  const isFirefox = browser.family === 'firefox'
+
+  // Secure is required for SameSite=None cookies to be set in secure contexts
+  // (https://w3c.github.io/webappsec-secure-contexts/#is-origin-trustworthy),
+  // but will not allow the cookie to be set in an insecure context.
+  // Normally http://localhost is considered a secure context (see
+  // https://w3c.github.io/webappsec-secure-contexts/#localhost), but Firefox
+  // does not consider the Cypress-launched browser to be a secure context (see
+  // https://github.com/cypress-io/cypress/issues/18217). For that reason, we
+  // remove Secure from http://localhost cookies in Firefox.
+  if (cookieSecureRegex.test(cookie)) {
+    if (isFirefox && isLocalhost && url.protocol === 'http:') {
+      debug('remove Secure from cookie')
+      cookie = cookie.replace(cookieSecureSemicolonRegex, '')
+    }
+  } else if (!isFirefox || url.protocol === 'https:') {
+    debug('add Secure to cookie')
+    cookie += '; Secure'
+  }
+
+  debug('resulting cookie: %s', cookie)
+
+  return cookie
+}
+
 const CopyCookiesFromIncomingRes: ResponseMiddleware = function () {
   const cookies: string | string[] | undefined = this.incomingRes.headers['set-cookie']
 
   if (cookies) {
-    ([] as string[]).concat(cookies).forEach((cookie) => {
+    const needsMultiDomainHandling = (
+      this.config.experimentalMultiDomain
+      && determineIfNeedsMultiDomainHandling(this)
+    )
+    const browser = this.getCurrentBrowser() || { family: null }
+    const url = new URL(this.req.proxiedUrl)
+    const isLocalhost = uri.isLocalhost(url)
+
+    debug('force SameSite=None?', needsMultiDomainHandling)
+
+    ;([] as string[]).concat(cookies).forEach((cookie) => {
```

Comment:
any reason the `;` is on line 456 instead of the end of 454? Did the linter do something weird?

Reply:
Starting a line with `[]` is pretty much the only case where you need a semicolon, because the interpreter thinks you might be using it for bracket notation based off the previous line (something like `(array || [])[0]`). I think it's conventional to put it at the beginning because it's the `[]` that causes the need for it, so if that line were removed, it wouldn't require changing another line to get rid of the then-unnecessary semicolon.